client/packages/host/src/components/Sync/SyncModal.tsx

```diff
@@ -137,203 +147,160 @@ export const SyncModal = ({
     syncStatus?.summary?.durationInSeconds || 0
   );
 
+  const getSyncStatusMessage = (): string => {
+    if (syncStatus?.isSyncing === true) {
+      return t('sync-info.syncing');
+    } else {
+      return numberOfRecordsInPushQueue
+        ? t('label.records-to-push', { count: numberOfRecordsInPushQueue })
+        : t('label.no-records-to-push');
+    }
+  };
+
+  const lastSuccessfulSyncMessage = (
+    date: Date,
+    durationAsDate: Date
+  ): string => {
+    // If the date is today, display the time, otherwise display the date
+    let today = new Date();
+    let lastSuccessfulSyncTime;
+    if (date?.toDateString() === today.toDateString()) {
+      lastSuccessfulSyncTime = localisedTime(date);
+    } else {
+      lastSuccessfulSyncTime = localisedDate(date);
+    }
+
+    // Format the duration into "X hours Y minutes Z seconds" omitting hours or minutes if there are 0 of them.
+    const hours = durationAsDate.getHours();
+    const minutes = durationAsDate.getMinutes();
+    const seconds = durationAsDate.getSeconds();
+    let formattedDuration = '';
+    if (hours > 0) {
+      formattedDuration += `${hours} ${t('label.hours', { count: hours })} `;
+    }
+    if (minutes > 0) {
+      formattedDuration += `${minutes} ${t('label.minutes', { count: minutes })} `;
+    }
+    formattedDuration += `${seconds} ${t('label.seconds', { count: seconds })}`;
+    formattedDuration = formattedDuration.trim();
+
+    // Return "Last successful sync 2:05 PM (completed in 1 second)"
+    return t('messages.last-successful-sync-time-and-duration', {
+      time: lastSuccessfulSyncTime,
+      duration: formattedDuration,
+    });
```

Comment:
Kind of like we did with the stocktake modal, i'd be tempted to move more of this to the localisations, to allow for difference ordering or syntax based on the current language. 

At least, changing the `label.hours` translation to display the `count` value that is passed in, rather than rendering it separately as you have above. Wondered about whether the whole thing could be one translation string, but that might be a bit too complex :)